packages/files/file.pony

```diff
@@ -1,4 +1,21 @@
-primitive _FileHandle
+use @_read[I32](fd: I32, buffer: Pointer[None], bytes_to_read: I32) if windows
+use @read[ISize](fd: I32, buffer: Pointer[None], bytes_to_read: USize) if not windows
+use @_write[I32](fd: I32, buffer: Pointer[None], bytes_to_send: I32) if windows
+use @writev[ISize](fd: I32, buffer: Pointer[None], num_to_send: I32) if not windows
+use @_lseeki64[I64](fd: I32, offset: I64, base: I32) if windows
+use @lseek[I64](fd: I32, offset: I64, base: I32) if not windows
```

Comment:
According to https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html setting `_FILE_OFFSET_BITS=64` will cause the 64 bit functions to be called.

>Macro: _FILE_OFFSET_BITS
>This macro determines which file system interface shall be used, one replacing the other. Whereas _LARGEFILE64_SOURCE makes the 64 bit interface available as an additional interface, _FILE_OFFSET_BITS allows the 64 bit interface to replace the old interface.
>
>If _FILE_OFFSET_BITS is undefined, or if it is defined to the value 32, nothing changes. The 32 bit interface is used and types like off_t have a size of 32 bits on 32 bit systems.
>
>If the macro is defined to the value 64, the large file interface replaces the old interface. I.e., the functions are not made available under different names (as they are with _LARGEFILE64_SOURCE). Instead the old function names now reference the new functions, e.g., a call to fseeko now indeed calls fseeko64.
>
>This macro should only be selected if the system provides mechanisms for handling large files. On 64 bit systems this macro has no effect since the *64 functions are identical to the normal functions.
>
>This macro was introduced as part of the Large File Support extension (LFS).

The indirect proof that this is working is the change required in `threads.c` for `rlim_t` on 32 bit platforms otherwise compiling fails because of a type mismatch.

If you'd like additional confirmation, I can compile a smaller C program with `_FILE_OFFSET_BITS=64` that calls `lseek` (or one of the other functions) and then check which symbols it is linked against.